### DISCUSSION TITLE
What is the correct way to reset data array in `useInfiniteQuery`

### DISCUSSION THREAD

**Discussion by marik22312** [TanStack/query#1628] [Q&A]
Hey, I have a table with an infinite scroll which uses `useInfiniteQuery` and basically always displayed the received data. (using react-query v2)

in some cases, I need to query my API with a new filter instead of a new page cursor.

for example
```ts
   // initial request
   {
    query: '',
   cursor: 0
  }

// next page request
{
  query: '',
  cursor: 1
}

// request after filtering
{
  query: 'someSearch',
  cursor: 0
}
```

I'm trying to reset the `data` array after querying with the new `query` key, so `data` will contain only the latest response from the API.

tried doing this with the `reset` and `clear` methods, but it doesn't seem to clear the data array...

**Comment by waptik** [comment]
Check [this](https://react-query.tanstack.com/reference/QueryClient#queryclientresetqueries) one out! In your case, you just have to pass the query key when calling.

Btw, how do you compose your query keys? Because if you do it better you won't need to reset queries.
